content/id/docs/concepts/overview/object-management-kubectl/declarative-config.md

```diff
@@ -0,0 +1,868 @@
+
+### Mengubah kepemilikan dari *writer* imperatif ke *file* konfigurasi
+
+Tambahkan *field* ke *file* konfigurasi. Hentikan pembaruan secara langsung ke konfigurasi *live* tanpa melalui `kubectl apply`.
+
+### Mengubah kepemilikan dari *file* konfigurasi ke *writer* imperatif
+
+Pada versi Kubernetes 1.5, mengubah kepemilikan sebuah field dari *file* konfigurasi memerlukan langkah-langkah manual:
+
+- Hapus *field* dari *file* konfigurasi.
+- Hapus *field* dari anotasi `kubectl.kubernetes.io/last-applied-configuration` pada objek *live*.
+
+## Mengubah metode-metode pengelolaan objek
+
+Objek-objek Kubernetes sebaiknya dikelola dengan satu metode dalam satu waktu. Berpindah dari satu metode ke metode lain dimungkinkan, tetapi memerlukan proses manual.
+
+{{< note >}}
+Boleh menggunakan perintah hapus secara imperatif dalam pengelolaan objek secara deklaratif.
+{{< /note >}}
+
+{{< comment >}}
+TODO(pwittrock): We need to make using imperative commands with
+declarative object configuration work so that it doesn't write the
+fields to the annotation, and instead.  Then add this bullet point.
+
+- using imperative commands with declarative configuration to manage where each manages different fields.
+{{< /comment >}}
+
+### Migrasi dari pengelolaan dengan perintah imperatif ke konfigurasi objek deklaratif
+
+Migrasi dari pengelolaan objek dengan perintah imperatif ke pengelolaan objek dengan konfigurasi deklaratif memerlukan beberapa langkah manual:
+
+1. Ekspor objek *live* ke *file* konfigurasi lokal:
+
+     ```shell
+     kubectl get <kind>/<name> -o yaml --export > <kind>_<name>.yaml
+     ```
+
+1. Hapus secara manual *field* `status` dari *file* konfigurasi.
+
+    {{< note >}}
+    Tahap ini opsional, karena `kubectl apply` tidak memperbarui *field* status meskipun *field* tersebut ada di *file* konfigurasi.
+    {{< /note >}}
+
+1. Ubah anotasi `kubectl.kubernetes.io/last-applied-configuration` pada objek:
+
+    ```shell
+    kubectl replace --save-config -f <kind>_<name>.yaml
+    ```
+
+1. Selanjutnya gunakan `kubectl apply` secara eksklusif untuk mengelola objek.
+
+{{< comment >}}
+TODO(pwittrock): Why doesn't export remove the status field?  Seems like it should.
+{{< /comment >}}
+
+### Migrasi dari konfigurasi objek imperatif ke konfigurasi objek deklaratif
+
+1. Atur anotasi `kubectl.kubernetes.io/last-applied-configuration` pada objek:
+
+    ```shell
+    kubectl replace --save-config -f <kind>_<name>.yaml
+    ```
+
+1. Gunakan selalu perintah `kubectl apply` saja untuk mengelola objek.
+
+## Pendefinisian selektor *controller* dan label PodTemplate
+
+{{< warning >}}
+Pembaruan selektor pada *controllers* sangat tidak disarankan.
+{{< /warning >}}
+
+Cara yang disarankan adalah dengan mendefinisikan sebuah PodTemplate *immutable* yang hanya digunakan oleh selektor *controller* tanpa memiliki arti semantik lainnya.
```

Comment:
Controller or kontroler?

Reply:
The word "kontroler" is not there yet in KBBI (https://kbbi.web.id).